ipynb/examples/trace_analysis/TraceAnalysis_IdleStates.ipynb

```diff
@@ -590,6 +623,125 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "## CPU idle state over time"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Take a look at the target's idle states:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "te.target.cpuidle.get_states()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now use trappy to plot the idle state of a single CPU over time. Higher is deeper: the plot is at -1 when the CPU is active, 0 for WFI, 1 for CPU sleep, etc.\n",
+    "\n",
+    "We should see that as the workload ramps down and the idle periods become longer, the idle states used become deeper."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "p = Parser(trace.ftrace, filters = {'cpu_id': cpu})\n",
+    "idle_df = p.solve('cpu_idle:state')\n",
+    "ILinePlot(idle_df, column=cpu, drawstyle='steps-post').view()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Examine idle period lengths\n",
+    "Let's get a DataFrame showing the length of each idle period on the CPU and the index of the cpuidle state that was entered."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def get_idle_periods(df):\n",
+    "    series = df[cpu]\n",
+    "    series = series[series.shift() != series].dropna()\n",
+    "    if series.iloc[0] == -1:\n",
+    "        series = series.iloc[1:]\n",
+    "\n",
+    "    idles = series.iloc[0::2] \n",
+    "    wakeups = series.iloc[1::2]\n",
+    "    if len(idles) > len(wakeups):\n",
+    "        idles = idles.iloc[:-1]\n",
+    "    else:\n",
+    "        wakeups = wakeups.iloc[:-1]\n",
+    "\n",
+    "    lengths = pd.Series((wakeups.index - idles.index), index=idles.index)\n",
+    "    return pd.DataFrame({\"length\": lengths, \"state\": idles})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make a scatter plot of the length of idle periods against the state that was entered. We should see that for long idle periods, deeper states were entered (i.e. we should see a positive corellation between the X and Y axes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "lengths = get_idle_periods(df)\n",
+    "lengths.plot(kind='scatter', x='length', y='state')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One more example: Draw a histogram of the length of idle periods shorter than 100ms in which the CPU entered cpuidle state 2"
```

Comment:
It would be good if this had a colon or period at the end of it (sorry, OCD :) ). Also, I don't believe it's necessary to add "One more example:", otherwise you'll have to add a potato at the end of your long list of examples :).